spec/factories/course_assessment_categories.rb

```diff
@@ -2,7 +2,7 @@
   sequence(:course_assessment_category_weight) { |n| n }
   sequence(:course_assessment_category_title) { |n| "Category #{n}" }
   factory :course_assessment_category, class: Course::Assessment::Category do
-    course
+    course { create(:course) }
```

Comment:
I'm using the component framework call back, so folder is created after course was created. https://github.com/Coursemology/coursemology2/blob/master/app/models/course/material/folder.rb#L20

But in category's before validation, course haven't been created, so there is no root folder that time.